scripts/apply-remote-migrations.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+app=${1}
+
+cd ./tdrs-backend
+
+echo "Install dependencies..."
+sudo apt install -y gcc
+sudo apt install -y libpq-dev python3-dev
+
+python -m venv ./env
+source ./env/bin/activate
+pip install --upgrade pip pipenv
+pipenv install --dev --system --deploy
+echo "Done."
+
+echo "Getting credentials..."
+guid=$(cf app --guid $app)
+app_vars=$(cf curl /v2/apps/$guid/env)
+
+db_creds=$(echo $app_vars | jq -r '.system_env_json.VCAP_SERVICES."aws-rds"[0].credentials')
+connection_str=$(echo $db_creds | jq -r '[.host, .port]' | jq -r 'join(":")')
+echo "Done."
+
+echo "Starting tunnel..."
+cf ssh -N -L 5432:$connection_str $app &
+sleep 5
+echo "Done."
+
+echo "Setting up environment..."
+cp ./.env.example ./.env.ci
+
+vcap_services=$(echo $app_vars | jq -r '.system_env_json.VCAP_SERVICES')
+vcap_application=$(echo $app_vars | jq -rc '.application_env_json.VCAP_APPLICATION')
+
+# replace host env var
+fixed_vcap_services=$(echo $vcap_services | jq -rc '."aws-rds"[0].credentials.host="localhost"')
+
+echo "VCAP_SERVICES='$fixed_vcap_services'" >> .env.ci
+echo "VCAP_APPLICATION='$vcap_application'" >> .env.ci
+
+set -a
+source .env.ci
+export DJANGO_CONFIGURATION=Development
+export DJANGO_SETTINGS_MODULE=tdpservice.settings.cloudgov
+set +a
+echo "Done."
+
+echo "Applying migrations..."
+# stop script and report errors??
+python manage.py migrate
```

Comment:
Wouldn't we need to exit this script with the return code here for a failure to be caught? Looking over the testing steps, I can't immediately come up with a way to test a failed migration.

Reply:
i've pushed up some changes that handle the exit code and end the ci task if appropriate
https://app.circleci.com/pipelines/github/raft-tech/TANF-app/20074/workflows/de105e6d-d857-4012-8886-11b43a068bc6/jobs/59734
![image](https://github.com/raft-tech/TANF-app/assets/111305129/68f377f1-d256-4390-8bfc-926996606471)

Reply:
Ok, this is looking better but perhaps using `set -e` to fail the script for any error? I'm not sure how significant this error from the pipeline really is but smells fishy:

```
[pipenv.exceptions.InstallError]: Successfully built awscli-local localstack-client
[pipenv.exceptions.InstallError]: Failed to build pygraphviz
[pipenv.exceptions.InstallError]: error: subprocess-exited-with-error
[pipenv.exceptions.InstallError]:   
[pipenv.exceptions.InstallError]:   × Building wheel for pygraphviz (pyproject.toml) did not run successfully.
[pipenv.exceptions.InstallError]:   │ exit code: 1
[pipenv.exceptions.InstallError]:   ╰─> [57 lines of output]
```

Reply:
`pygraphviz` requires `graphviz` in order to build, which we removed. in general i think catching the return of `python manage.py migrate` here should be sufficient - if anything preventing migrations from running fails to build, it should result in `migrate` failing as well. if not, well `migrations` should have run and we should be good. 

i'm open to other perspectives here though

Reply:
I'm ok with it just a thought and it does catch bad migrations per [my testing](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/20235/workflows/99e17ba9-244d-4752-958f-7cd304872ec9/jobs/60221)